satpy/writers/__init__.py

```diff
@@ -325,17 +327,44 @@ def get_filename(self, **kwargs):
                 "No filename pattern or specific filename provided")
         return self.filename_parser.compose(kwargs)
 
-    def save_datasets(self, datasets, **kwargs):
+    def save_datasets(self, datasets, compute=True, **kwargs):
         """Save all datasets to one or more files.
 
         Subclasses can use this method to save all datasets to one single
         file or optimize the writing of individual datasets. By default
         this simply calls `save_dataset` for each dataset provided.
         """
+        sources = []
+        targets = []
         for ds in datasets:
-            self.save_dataset(ds, **kwargs)
+            res = self.save_dataset(ds, compute=False, **kwargs)
+            if isinstance(res, tuple):
```

Comment:
I can't see `self.save_dataset` returning a tuple, even longer down the call chain, am I missing something ?